File: src/test/java/com/ject/studytrip/auth/fixture/KakaoSignupRequestFixture.java
package com.ject.studytrip.auth.fixture;

import com.ject.studytrip.auth.presentation.dto.request.KakaoSignupRequest;

public class KakaoSignupRequestFixture {
    private String category = "STUDENT";
    private String nickname = "민우";

    public KakaoSignupRequestFixture withCategory(String category) {
        this.category = category;
        return this;
    }

    public KakaoSignupRequestFixture withNickname(String nickname) {
        this.nickname = nickname;
        return this;
    }

    public KakaoSignupRequest build() {
        return new KakaoSignupRequest(category, nickname);
    }
}

File: src/test/kotlin/com/ject/studytrip/auth/fixture/KakaoSignupRequestFixture.kt
package com.ject.studytrip.auth.fixture

import com.ject.studytrip.auth.presentation.dto.request.KakaoSignupRequest

class KakaoSignupRequestFixture(
    private val category: String = "STUDENT",
    private val nickname: String = "민우",
) {
    fun build(): KakaoSignupRequest = KakaoSignupRequest(category, nickname)
}
